### PR TITLE
Redesigning sim_telarray runners

### PR DESCRIPTION
**Pull request by RaulRPrado**
This PR separates the class SimtelRunner into a base class with the same name and a sub class SimtelRunnerRayTracing.

Previously, the class SimtelRunner contained different 'modes', which was not too clean.

Next, another sub class will be created to deal with array simulations, SimtelRunnerArray.

### CODE REVIEW

File: simtools/simtel/simtel_runner_ray_tracing.py [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+
+        self._loadRequiredFiles()
+
+    def _loadRequiredFiles(self):
+        '''
+        Which file are required for running depends on the mode.
+        Here we define and write some information into these files. Log files are always requires.
+        '''
+
+        self._corsikaFile = self._simtelSourcePath.joinpath('run9991.corsika.gz')
```
Comment (GernotMaier) [inline simtools/simtel/simtel_runner_ray_tracing.py:126]: Not sure I get this: file name is hard wired here?

File: simtools/simtel/simtel_runner_ray_tracing.py [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+        # Loop to define and remove existing files.
+        # Files will be named _baseFile = self.__dict__['_' + base + 'File']
+        for baseName in ['stars', 'photons', 'log']:
+            fileName = names.rayTracingFileName(
+                self.telescopeModel.site,
+                self.telescopeModel.name,
+                self.config.sourceDistance,
+                self.config.zenithAngle,
+                self.config.offAxisAngle,
+                self.config.mirrorNumber if self._singleMirrorMode else None,
```
Comment (GernotMaier) [inline simtools/simtel/simtel_runner_ray_tracing.py:137]: Does that mean one cannot run >1 mirror telescope (I like misunderstand the line)

File: simtools/simtel/simtel_runner_ray_tracing.py [hunk 3]
```diff
@@ -0,0 +1,256 @@
...
+            file.write('# zenithAngle [deg] = {}\n'.format(self.config.zenithAngle))
+            file.write('# offAxisAngle [deg] = {}\n'.format(self.config.offAxisAngle))
+            file.write('# sourceDistance [km] = {}\n'.format(self.config.sourceDistance))
+            if self._singleMirrorMode:
+                file.write('# mirrorNumber = {}\n\n'.format(self.config.mirrorNumber))
+
+        # Filling in star file with a single star.
+        with self._starsFile.open('w') as file:
+            file.write('0. {} 1.0 {}'.format(
+                90. - self.config.zenithAngle,
```
Comment (GernotMaier) [inline simtools/simtel/simtel_runner_ray_tracing.py:162]: You explained it in the past to me, but just to be sure: you ensure the units somewhere else (e.g., that angles are in degrees)

**Review by GernotMaier — APPROVED** [review]

File: simtools/simtel/simtel_runner_ray_tracing.py (hunk 1 above)
Comment (RaulRPrado) [reply simtools/simtel/simtel_runner_ray_tracing.py:126]: Yes, this is a empty CORSIKA file that used to come in the sim_telarray package. I realized recently that it is actually not there for the recent versions.
The file not existing is not a problem for the ray tracing simulations itself. The only problem is that the sim_telarray execution returns an error code, so we can't actually identify real execution errors.

I created an issue to go back to this later.

File: simtools/simtel/simtel_runner_ray_tracing.py (hunk 2 above)
Comment (RaulRPrado) [reply simtools/simtel/simtel_runner_ray_tracing.py:137]: No, one can run the full dish or single mirror telescopes. Why? I don't get it.

File: simtools/simtel/simtel_runner_ray_tracing.py (hunk 3 above)
Comment (RaulRPrado) [reply simtools/simtel/simtel_runner_ray_tracing.py:162]: Yes. This is done by our system to validate configurable parameters. Any parameter in self.config was validated and converted to the proper unit.

**Comment by RaulRPrado** [comment]
I will merge it but we can continue the discussions.

File: simtools/simtel/simtel_runner_ray_tracing.py (hunk 2 above)
Comment (GernotMaier) [reply simtools/simtel/simtel_runner_ray_tracing.py:137]: Wait: single mirror == 1 mirror panel? I though we discuss here single vs dual-mirror (dish) telescopes.

I guess we need to have a clear understanding of the language, mirror is really unclear.

Comment (RaulRPrado) [reply simtools/simtel/simtel_runner_ray_tracing.py:137]: Yes, I'm talking about mirror panels, not dishes.

**Pull request merged**